src/Query/Builder.php

```diff
@@ -647,6 +647,29 @@ public function increment($column, $amount = 1, array $extra = [], array $option
         return $this->performUpdate($query, $options);
     }
 
+    public function incrementEach(array $columns, array $extra = [], array $options = [])
+    {
+
+        $query = ['$inc' => $columns];
+
+        if (! empty($extra)) {
+            $query['$set'] = $extra;
+        }
+
+        // Protect
+        foreach ($columns as $column => $amount) {
+            $this->where(function ($query) use ($column) {
+                $query->where($column, 'exists', false);
+
+                $query->orWhereNotNull($column);
```

Comment:
With cumulative conditions on all fields, no field will be incremented if at least 1 of the fields is `null`.

We'd be wise not to set any conditions at all and let the server return an error.